### DISCUSSION TITLE
How to judge a function is MST action

### DISCUSSION THREAD

**Discussion by BanShan-Alec** [mobxjs/mobx-state-tree#2046] [General]
when I use mobx-state-tree，I need judge a function is MST action. After reading MST Document，I can not find a api to resolve it.

So I make a question issue in mbox repo, they suggest me to seek help at here.

https://github.com/mobxjs/mobx/issues/3726

**Comment by coolsoftwaretyler** [comment]
Hey @BanShan-Alec - I have a few thoughts. I found your [CodeSandbox](https://codesandbox.io/s/mobx-state-tree-playground-forked-yxhy6p?file=/src/index.js) and have been playing around with it.

1. MobX folks are correct, I don't think you should expect `isAction` to work on MST actions.
2. I don't think MST has a great API for what you're trying to do. In the meantime, you could always use `typeof` to check if something is a function. Kind of like this: 

```js
function isValidAction(actionName, model) {
  return typeof model[actionName] === "function";
}
```

Of course, that won't give you the precise answer, since other items on the model instance might be functions as well, such as views.

You mentioned in the MobX discussion that you are using some MST internals as a workaround. Would you be willing to: 

1. Share your use case a little more specifically (what are you trying to accomplish and why)
2. Share the code you're currently using to solve the problem.

I'd definitely be interested in a PR to add something like this. I have been unable to find a really good out-of-the-box solution as-is.

Final note here, I think this might be somewhat related to a longstanding issue we closed out about [distinguishing actions and properties](https://github.com/mobxjs/mobx-state-tree/issues/1963). There is an underlying philosophy here to keep things simple, so I don't think we need to rework anything, but if you found a path forward from reading an internal, there's probably a way for us to write a stable API to surface that information.

**Comment by coolsoftwaretyler** [comment]
I'm going to go ahead and mark this as closed since it's been a while without a response. Feel free to re-open!
